app/models/post_upload.rb

```diff
@@ -1,21 +0,0 @@
-# frozen_string_literal: true
```

Comment:
Still don't think this model should be deleted in this PR, it is referred to in a few plugin places:

```
all-the-custom-plugins/plugins/discourse-warner-dc/db/fixtures/001_badges.rb
73:        INNER JOIN post_uploads pu

all-the-plugins/plugins/discourse-data-explorer/lib/queries.rb
351:    FROM post_uploads
352:    JOIN uploads ON uploads.id = post_uploads.upload_id
353:    JOIN posts ON posts.id = post_uploads.post_id

all-the-plugins/plugins/discourse-encrypt/spec/lib/email_sender_spec.rb
43:      reply.link_post_uploads
82:      reply.link_post_uploads

all-the-plugins/plugins/discourse-chat/spec/lib/chat_channel_archive_service_spec.rb
125:        expect(@channel_archive.destination_topic.posts.last.post_uploads.count).to eq(1)
```

We can just add a note that it's no longer used (pointing to `UploadReference`) and delete it when we drop the table later after plugins are changed.